java/src/main/java/com/genexus/util/GXStorageProvider.java

```diff
@@ -50,7 +51,7 @@ public boolean uploadPrivate(String filefullpath, String storageobjectfullname,
             if (storageobjectfullname.isEmpty()) {
                 storageobjectfullname = new File(filefullpath).getName();
             }
-            String url = provider.upload(filefullpath, storageobjectfullname, true);
+            String url = provider.upload(filefullpath, storageobjectfullname, ResourceAccessControlList.Private);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 49:62, 32:45.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.